Authentication/SignUp/View/SignupView.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  SignupViewType.swift
+//  Authentication
+//
+//  Created by Daniela Riesgo on 3/18/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+import Foundation
+
+public protocol SignupViewType: Renderable {
+ 
+    var view: UIView { get }
+    
+    var usernameLabel: UILabel? { get }
+    var usernameTextField: UITextField? { get }
+    var usernameValidationMessageLabel: UILabel? { get }
+    
+    var emailLabel: UILabel? { get }
+    var emailTextField: UITextField { get }
+    var emailValidationMessageLabel: UILabel? { get }
+    
+    var passwordLabel: UILabel? { get }
+    var passwordTextField: UITextField { get }
+    var passwordValidationMessageLabel: UILabel? { get }
+    var passwordVisibilityButton: UIButton? { get }
+    
+    var passwordConfirmLabel: UILabel? { get }
+    var passwordConfirmTextField: UITextField { get }
+    var passwordConfirmValidationMessageLabel: UILabel? { get }
+    var passwordConfirmVisibilityButton: UIButton? { get }
+    
+    var signupButton: UIButton { get }
+    var signupErrorLabel: UILabel? { get }
+    
+    var termsAndServicesButton: UIButton { get }
+    var termsAndServicesLabel: UILabel? { get }
+    
+    var usernameTextFieldValid: Bool { get set }
+    var usernameTextFieldSelected: Bool { get set }
+    var emailTextFieldValid: Bool { get set }
+    var emailTextFieldSelected: Bool { get set }
+    var passwordTextFieldValid: Bool { get set }
+    var passwordTextFieldSelected: Bool { get set }
+    var passwordConfirmationTextFieldValid: Bool { get set }
+    var passwordConfirmationTextFieldSelected: Bool { get set }
+    var signupButtonEnabled: Bool { get set }
+    var signupButtonPressed: Bool { get set }
+    
```

Comment:
What I would do is make the Form types include all textfields and labels and maybe the main button. But leave the logo and the buttons to other screens to the LoginViewType or SignupViewType.